ethToElrond/v2/ethToElrond/interface.go

```diff
@@ -0,0 +1,30 @@
+package ethToElrond
+
+import (
+	"context"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/clients"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+// EthToElrondBridge defines the operations for the Ethereum to Elrond half-bridge
+type EthToElrondBridge interface {
+	GetLogger() logger.Logger
+	MyTurnAsLeader() bool
```

Comment:
we have something similar on elrond-go(consensus). For simplicity I think it is better to keep it like that ftm also because this can be easily adapted with your propose in the future if more role will come into play.